core/src/apps/ethereum/sign_tx.py

```diff
@@ -165,13 +165,19 @@ def check(msg: EthereumSignTx):
     if msg.tx_type not in [1, 6, None]:
         raise wire.DataError("tx_type out of bounds")
 
+    check_chain_id(msg)
     check_data(msg)
 
     # safety checks
     if not check_gas(msg) or not check_to(msg):
         raise wire.DataError("Safety check failed")
 
 
+def check_chain_id(msg: EthereumSignTx):
```

Comment:
After reviewing the corresponding code in legacy, I don't think this is necessary (and actually modifies a feature).

When `chain_id` is 0 or unset, it is not hashed in (per https://github.com/trezor/trezor-firmware/pull/1771/files#R75), producing a pre-EIP155 signature. This check would disable pre-EIP155 signatures.